src/components/exit-this-page/index.md

```diff
@@ -0,0 +1,104 @@
+
+When the user presses the button, their browser will immediately go to a predetermined page that’s set in the code, which is BBC weather by default.
+
+### Choosing a different website to load
+
+The component is set to take the user to the [homepage of BBC Weather](https://www.bbc.co.uk/weather). You can change this if there’s a more appropriate site for your users.
+
+Avoid websites that might show personalised pages (such as frequently visited, last visited, or suggested links), as this content could put the user at risk.
+
+## Consider what to do user session data
+
+You’ll need to decide if your service will store the user’s session data after they’ve activated the Exit this page component and left your service.
+
+We recommend that services clear the user’s session data once they’ve activated Exit this page. This can be through the button, the secondary skip link or pressing the shift button 3 times.
+
+## Adding the secondary link
+
+The code will need to go in the layout file of your service. It will need to go after your block header.
+
+To activate the secondary link, users will need to either:
+- tab to the Exit this page link
+- press the shift key 3 times
+
+These will perform the same action as pressing the button.
+
+### Showing progress for 'Shift key' presses
+
+To show the user the number of times they’ve pressed the shift button, there’s a series of 3 progress dots under the button, provided as a visual enhancement. Each time the shift key is pressed, one of the dots will be filled.
+
+The user has 5 seconds to press the shift key 3 times. After that, the sequence and progress dots will reset.
+
+This feature requires the user to have Javascript enabled on their browser.
+
+## Loading overlay
+
+When the user activates Exit this page, a ‘loading overlay’ will immediately appear. This is to help users with slow connections clear their screen until the browser loads the next page.
+
+## Research on this component
+
+The design of this component is based on research with people with a lived experience of domestic abuse and people with accessibility needs, and in consultation with the Ministry of Justice, Department for Work and Pensions and the Scottish Government.
+
+Live examples of similar components can be found on these websites:
+- [GOV.UK: Check if you can get legal aid](https://www.gov.uk/check-legal-aid)
+- [GOV.UK: Apply for help arranging child maintenance](https://child-maintenance.service.gov.uk/apply/eligibility/info)
+- [mygov.scot: Domestic abuse support](https://www.mygov.scot/domestic-abuse/)
+- [Women’s aid website](https://www.womensaid.org.uk/)
+- [Refuge website](https://www.nationaldahelpline.org.uk/)
+
+### Accessibility features and considerations
+
+We'll explain why there’s the keyboard option and the skip-link version.
+
+We'll explain our testing with Dragon, and what we expect to be possible.
```

Comment:
Need to fill in, possibly this can be adapted from blog drafts

```suggestion
We'll explain why there’s the keyboard option and the skip-link version.

We'll explain our testing with Dragon, and what we expect to be possible.
```

Reply:
Outline:
- there’s 2 additional ways to activate EtP
- keyboard option for keyboard users with motor impairments, also offers a discreet options
- ...some iterations use single key press, but we were not confident that this could be done without interfering with other user actions
- skip link option for users to give assistive tech users an easier way to reliably reach for the button
- ...investigated a disguised command, but decided that ‘item numbers’, etc. was sufficient
- and then a summary of accessibility testing? (Can someone send me a link and summarise findings?)

Reply:
First draft at this. Still need a link or summary of accessibility testing.

```suggestion
In addition to the 'Exit this page' button itself, users have 2 other options to activate the component. We've tested and iterated both of these options to be accessible, reliable and discreet as possible for users.

The keyboard option, where the user can press shift 3 times, gives users a discreet way to activate the component, particularly keyboard-only users. We considered options that used a single key press (such as the 'Esc' key), but we could not be confident that this would not interfere with other user actions.

The 'skip link' option is a hidden link in the tab order of the page, immediately between [xxxxx] and [xxxxx]. This gives users of assistive technology and easier way to reliable activate the component. 

To help users of speech recognition software activate the 'skip link' more discreetly, we considered disguising the 'Exit this page' link text with something else. However, we decided that this could confuse users and other methods, such as referring to the link by item number would meet their need.
```